Documentation/maintenance.md

```diff
@@ -0,0 +1,139 @@
+Depending on an etcd application's needs, this maintenance can usually
+be automated and performed without down time or significantly degraded performance.
+
+All etcd maintenance manages storage resources consumed by the etcd keyspace.
+Failure to adequately control the keyspace size is guarded by storage space quotas;
+if an etcd member runs low on space, a quota will trigger cluster-wide alarms
+which will put the system into a limited-operation maintenance mode that
+forbids writing new key data.
+To avoid running out of space for writes to the keyspace, the etcd keyspace
+history must be regularly compacted as it ingests writes.
+Storage space itself may be reclaimed by defragmenting etcd members.
+Finally, periodic snapshot backups of etcd member state makes it possible to recover any unintended logical data loss or corruption caused by operational error.
+
+## History Compaction
+
+Since etcd keeps an exact history of its keyspace, this history
+should be periodically compacted to avoid performance degradation and eventual storage space exhaustion. Compacting the keyspace history drops all information about keys superseded prior to 
+a given keyspace revision. The space used by these keys then becomes available for additional writes to the keyspace.
+
+Compaction may be invoked automatically through `etcd`'s time-based compaction retention policy
+or through`etcdctl`.
+The `etcdctl` approach is useful for fine-grained control over compaction whereas automatic compaction addresses applications that only need time-bounded history.
+
+Automatic compaction is activated by passing a command line flag to `etcd`:\
+```sh
+# keep one hour of history
+$ etcd --experimental-auto-compaction-retention=1
+```
+
+An `etcdctl` initiated compaction works as follows:
+```sh
+# compact up to revision 3
+$ etcdctl compact 3
+
+```
+
+Note that keys in old revisions are inaccessible:
+```sh
+$ etcdctl get --rev=2 somekey
+Error:  rpc error: code = 11 desc = etcdserver: storage: required revision has been compacted
+```
+
+## Defragmentation
+
+After compacting the keyspace, the backend database may exhibit internal fragmentation.
+Any internal fragmentation is space free to use by the backend for future writes but still
+consumes storage space. The process of defragmentation releases this storage space
+back to the file system. Defragmentation blocks keyspace requests while running; it is issued
+on a per-member basis so that cluster-wide latency spikes may be avoided.
+
+To defragment an etcd member, use the `etcdctl defrag` command:
+```sh
+$ etcdctl defrag
+Finished defragmenting etcd member[127.0.0.1:2379]
+```
+
+## Space Quota
+
+The space quota in `etcd` ensures the cluster only fully operates in a reliable fashion.
+Without a space quota, the `etcd` may suffer from poor performance by having an excessively
+large keyspace or simply run out of space, leading to unpredictable system behavior.
```

Comment:
How about:
"Without a space quota, `etcd` may suffer poor performance when the keyspace grows excessively large, or it may simply run entirely out of storage space, leading to unpredictable behavior."